### PR TITLE
add bella rgb keyboards

### PR DESCRIPTION
**Pull request by moyi4681**
add bella rgb keyboards

### CODE REVIEW

File: keyboards/kbdfans/bella/rgb/rules.mk [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/kbdfans/bella/rgb/rules.mk:41]:
```suggestion
# Bootloader
BOOTLOADER = atmel-dfu
```

File: keyboards/kbdfans/bella/rgb_iso/rules.mk [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/kbdfans/bella/rgb_iso/rules.mk:41]:
```suggestion
# Bootloader
BOOTLOADER = atmel-dfu
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kbdfans/bella/rgb/info.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "bella_rgb", 
+    "url": "", 
+    "maintainer": "moyi4681", 
+    "width": 16.25, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/kbdfans/bella/rgb/info.json:8]: May need further changes, but at a minimum
```suggestion
        "LAYOUT_all": {
```

File: keyboards/kbdfans/bella/rgb_iso/info.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "bella_rgb_iso", 
+    "url": "", 
+    "maintainer": "moyi4681", 
+    "width": 16.25, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/kbdfans/bella/rgb_iso/info.json:8]: May need further changes, but at a minimum
```suggestion
        "LAYOUT_all": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/bella/rgb/config.h [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+//#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_SOLID_COLOR
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_ADDR_2 0b0110000 // this is here for compliancy reasons.
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 108
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#define DRIVER_INDICATOR_LED_TOTAL 0
+#endif
+#define VIA_EEPROM_LAYOUT_OPTIONS_SIZE 2
```
Comment (zvecr) [inline keyboards/kbdfans/bella/rgb/config.h:57]:
```suggestion
#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```

File: keyboards/kbdfans/bella/rgb_iso/config.h [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+//#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_SOLID_COLOR
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_ADDR_2 0b0110000 // this is here for compliancy reasons.
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 109
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#define DRIVER_INDICATOR_LED_TOTAL 0
+#define VIA_EEPROM_LAYOUT_OPTIONS_SIZE 2
+#endif
```
Comment (zvecr) [inline keyboards/kbdfans/bella/rgb_iso/config.h:57]:
```suggestion
#endif
#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/bella/rgb_iso/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+AUDIO_ENABLE = no              # Audio output
+RGB_MATRIX_ENABLE = IS31FL3741 # Use RGB matrix
```
Comment (fauxpark) [inline keyboards/kbdfans/bella/rgb_iso/rules.mk:20]:
```suggestion
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = IS31FL3741
```

File: keyboards/kbdfans/bella/rgb/rules.mk [hunk 8]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader
```
Comment (fauxpark) [inline keyboards/kbdfans/bella/rgb/rules.mk:4]:
```suggestion
# Bootloader selection
```

File: keyboards/kbdfans/bella/rgb/rules.mk [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+AUDIO_ENABLE = no              # Audio output
+RGB_MATRIX_ENABLE = IS31FL3741 # Use RGB matrix
```
Comment (fauxpark) [inline keyboards/kbdfans/bella/rgb/rules.mk:20]:
```suggestion
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = IS31FL3741
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/bella/rgb/rules.mk [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = lite        # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/kbdfans/bella/rgb/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite        # Virtual DIP switch configuration
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kbdfans/bella/rgb/readme.md [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+* Keyboard Maintainer: [DZTECH](https://github.com/moyi4681)
+* Hardware Supported: BELLA_RGB
+* Hardware Availability: [KBDFans](https://kbdfans.cn/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kbdfans/bella/rgb:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb/readme.md:11]:
```suggestion
Flashing example for this keyboard:

    make kbdfans/bella/rgb:default:flash

## Accessing Bootloader Mode

To reset the board into bootloader mode, hold the key at the top left of the keyboard while connecting the USB cable (also erases persistent settings).

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/kbdfans/bella/rgb_iso/config.h [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+ */
+#pragma once
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B42
+#define PRODUCT_ID      0x0009
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    KBDFANS
+#define PRODUCT         BELLA_RGB_ISO
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/config.h:24]:
```suggestion
#define PRODUCT         BELLA RGB ISO
```

File: keyboards/kbdfans/bella/rgb_iso/rgb_iso.h [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_all( \
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/rgb_iso.h:20]:
```suggestion
#define LAYOUT( \
```

File: keyboards/kbdfans/bella/rgb_iso/rgb_iso.h [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+	K000,       K002, K003, K004, K005, K006, K007, K008, K009,       K011, K012, K013, K014, K015, \
+	K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112,       K114, K115, \
+	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K313,       K215, \
+	K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311,       K213,       K315, \
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/rgb_iso.h:24]: Moves the ISO Enter key to the home row in the keymaps.

```suggestion
	K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212,             K215, \
	K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, K213, K313,       K315, \
```

File: keyboards/kbdfans/bella/rgb_iso/info.json [hunk 15]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "bella_rgb_iso",
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/info.json:2]:
```suggestion
    "keyboard_name": "BELLA RGB ISO", 
```

File: keyboards/kbdfans/bella/rgb_iso/info.json [hunk 16]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "bella_rgb_iso", 
+    "url": "", 
+    "maintainer": "moyi4681", 
+    "width": 16.25, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/info.json:8]:
```suggestion
        "LAYOUT": {
```

File: keyboards/kbdfans/bella/rgb_iso/info.json [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
... "x":3.75, "y":3}, {"label":"F", "x":4.75, "y":3}, {"label":"G", "x":5.75, "y":3}, {"label":"H", "x":6.75, "y":3}, {"label":"J", "x":7.75, "y":3}, {"label":"K", "x":8.75, "y":3}, {"label":"L", "x":9.75, "y":3}, {"label":":", "x":10.75, "y":3}, {"label":"\"", "x":11.75, "y":3}, {"label":"~", "x":12.75, "y":3}, {"label":"Page Down", "x":15.25, "y":3}, {"label":"Shift", "x":0, "y":4, "w":1.25}, {"label":"|", "x":1.25, "y":4}, {"label":"Z", "x":2.25, "y":4}, {"label":"X", "x":3.25, "y":4}, {"label":"C", "x":4.25, "y":4}, {"label":"V", "x":5.25, "y":4}, {"label":"B", "x":6.25, "y":4}, {"label":"N", "x":7.25, "y":4}, {"label":"M", "x":8.25, "y":4}, {"label":"<", "x":9.25, "y":4}, {"label":">", "x":10.25, "y":4}, {"label":"?", "x":11.25, "y":4}, {"label":"Shift", "x":12.25, "y":4, "w":1.75}, {"label":"\u2191", "x":14.25, "y":4.25}, {"label":"Ctrl", "x":0, "y":5, "w":1.25}, {"label":"Win", "x":1.25, "y":5, "w":1.25}, {"label":"Alt", "x":2.5, "y":5, "w":1.25}, {"x":3.75, "y":5, "w":6.25}, {"label":"Alt", "x":10, "y":5, "w":1.5}, {"label":"Fn", "x":11.5, "y":5, "w":1.5}, {"label":"\u2190", "x":13.25, "y":5.25}, {"label":"\u2193", "x":14.25, "y":5.25}, {"label":"\u2192", "x":15.25, "y":5.25}]
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/info.json:9]:
```suggestion
            "layout": [{"label":"Esc", "x":0, "y":0}, {"label":"F1", "x":2, "y":0}, {"label":"F2", "x":3, "y":0}, {"label":"F3", "x":4, "y":0}, {"label":"F4", "x":5, "y":0}, {"label":"F5", "x":6.5, "y":0}, {"label":"F6", "x":7.5, "y":0}, {"label":"F7", "x":8.5, "y":0}, {"label":"F8", "x":9.5, "y":0}, {"label":"F9", "x":11, "y":0}, {"label":"F10", "x":12, "y":0}, {"label":"F11", "x":13, "y":0}, {"label":"F12", "x":14, "y":0}, {"label":"Page Up", "x":15.25, "y":0}, {"label":"~", "x":0, "y":1}, {"label":"!", "x":1, "y":1}, {"label":"@", "x":2, "y":1}, {"label":"#", "x":3, "y":1}, {"label":"$", "x":4, "y":1}, {"label":"%", "x":5, "y":1}, {"label":"^", "x":6, "y":1}, {"label":"&", "x":7, "y":1}, {"label":"*", "x":8, "y":1}, {"label":"(", "x":9, "y":1}, {"label":")", "x":10, "y":1}, {"label":"_", "x":11, "y":1}, {"label":"+", "x":12, "y":1}, {"label":"Backspace", "x":13, "y":1, "w":2}, {"label":"Insert", "x":15.25, "y":1}, {"label":"Tab", "x":0, "y":2, "w":1.5}, {"label":"Q", "x":1.5, "y":2}, {"label":"W", "x":2.5, "y":2}, {"label":"E", "x":3.5, "y":2}, {"label":"R", "x":4.5, "y":2}, {"label":"T", "x":5.5, "y":2}, {"label":"Y", "x":6.5, "y":2}, {"label":"U", "x":7.5, "y":2}, {"label":"I", "x":8.5, "y":2}, {"label":"O", "x":9.5, "y":2}, {"label":"P", "x":10.5, "y":2}, {"label":"{", "x":11.5, "y":2}, {"label":"}", "x":12.5, "y":2}, {"label":"Delete", "x":15.25, "y":2}, {"label":"Caps Lock", "x":0, "y":3, "w":1.75}, {"label":"A", "x":1.75, "y":3}, {"label":"S", "x":2.75, "y":3}, {"label":"D", "x":3.75, "y":3}, {"label":"F", "x":4.75, "y":3}, {"label":"G", "x":5.75, "y":3}, {"label":"H", "x":6.75, "y":3}, {"label":"J", "x":7.75, "y":3}, {"label":"K", "x":8.75, "y":3}, {"label":"L", "x":9.75, "y":3}, {"label":":", "x":10.75, "y":3}, {"label":"\"", "x":11.75, "y":3}, {"label":"~", "x":12.75, "y":3}, {"label":"Enter", "x":13.75, "y":2, "w":1.25, "h":2}, {"label":"Page Down", "x":15.25, "y":3}, {"label":"Shift", "x":0, "y":4, "w":1.25}, {"label":"|", "x":1.25, "y":4}, {"label":"Z", "x":2.25, "y":4}, {"label":"X", "x":3.25, "y":4}, {"label":"C", "x":4.25, "y":4}, {"label":"V", "x":5.25, "y":4}, {"label":"B", "x":6.25, "y":4}, {"label":"N", "x":7.25, "y":4}, {"label":"M", "x":8.25, "y":4}, {"label":"<", "x":9.25, "y":4}, {"label":">", "x":10.25, "y":4}, {"label":"?", "x":11.25, "y":4}, {"label":"Shift", "x":12.25, "y":4, "w":1.75}, {"label":"\u2191", "x":14.25, "y":4.25}, {"label":"Ctrl", "x":0, "y":5, "w":1.25}, {"label":"Win", "x":1.25, "y":5, "w":1.25}, {"label":"Alt", "x":2.5, "y":5, "w":1.25}, {"x":3.75, "y":5, "w":6.25}, {"label":"Alt", "x":10, "y":5, "w":1.5}, {"label":"Fn", "x":11.5, "y":5, "w":1.5}, {"label":"\u2190", "x":13.25, "y":5.25}, {"label":"\u2193", "x":14.25, "y":5.25}, {"label":"\u2192", "x":15.25, "y":5.25}]
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c:20]:
```suggestion
	LAYOUT(
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+		KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_PGUP, 
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,KC_BSPC, KC_INS, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_ENT, KC_DEL, 
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_PGDN,
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c:24]:
```suggestion
		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,         KC_DEL, 
		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT, KC_PGDN, 
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+
+	LAYOUT_all(
+		KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_PGUP, 
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,KC_BSPC, KC_INS, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_ENT, KC_DEL, 
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_PGDN, 
+		KC_LSFT, KC_NUBS,KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_RALT, MO(1), KC_LEFT, KC_DOWN, KC_RGHT),
+
+	LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c:28]:
```suggestion
	LAYOUT(
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, RGB_MOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, EEP_RST, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS,
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/default/keymap.c:32]:
```suggestion
		KC_TRNS, RGB_MOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, 
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, EEP_RST, KC_TRNS, 
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c:20]:
```suggestion
	LAYOUT(
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+		KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_PGUP, 
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,KC_BSPC, KC_INS, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_ENT, KC_DEL, 
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_PGDN,
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c:24]:
```suggestion
		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,        KC_DEL, 
		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT, KC_PGDN, 
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+	LAYOUT_all(
+		KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_PGUP, 
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,KC_BSPC, KC_INS, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_ENT, KC_DEL, 
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_BSLS, KC_PGDN, 
+		KC_LSFT, KC_NUBS,KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_RALT, MO(1), KC_LEFT, KC_DOWN, KC_RGHT),
+
+	LAYOUT_all(
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c:28]:
```suggestion
	LAYOUT(
```

File: keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, RGB_MOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, EEP_RST, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,   KC_TRNS,
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/keymaps/via/keymap.c:32]:
```suggestion
		KC_TRNS, RGB_MOD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, 
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,   EEP_RST, KC_TRNS, 
```

File: keyboards/kbdfans/bella/rgb_iso/readme.md [hunk 26]
```diff
@@ -0,0 +1,11 @@
+# BELLA_RGB_ISO (Hotswap)
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/readme.md:1]:
```suggestion
# BELLA RGB ISO (Hotswap)
```

File: keyboards/kbdfans/bella/rgb_iso/readme.md [hunk 27]
```diff
@@ -0,0 +1,11 @@
...
+
+* Keyboard Maintainer: [DZTECH](https://github.com/moyi4681)
+* Hardware Supported: BELLA_RGB_ISO
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/readme.md:4]:
```suggestion
* Hardware Supported: BELLA RGB ISO (ATmega32U4)
```

File: keyboards/kbdfans/bella/rgb_iso/readme.md [hunk 28]
```diff
@@ -0,0 +1,11 @@
...
+* Hardware Availability: [KBDFans](https://kbdfans.cn/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kbdfans/bella/rgb_iso:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (noroadsleft) [inline keyboards/kbdfans/bella/rgb_iso/readme.md:11]:
```suggestion
Flashing example for this keyboard:

    make kbdfans/bella/rgb_iso:default:flash

## Accessing Bootloader Mode

To reset the board into bootloader mode, hold the key at the top left of the keyboard while connecting the USB cable (also erases persistent settings).

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/11438/files).

**Comment by noroadsleft** [comment]
@moyi4681 I took the liberty of applying the remaining suggestions I had left that you missed to your pull request, because they were the only things remaining to make it merge-ready, and the compilation wouldn't have worked without them.

Thanks!

**Pull request merged**